src/resources/views/crud/fields/select2.blade.php

```diff
@@ -1,76 +1,158 @@
+
+        $refreshRoute = route($createRouteEntity."-on-the-fly-refresh-options");
+
+    }else{
+        $activeOnTheFlyCreate = false;
+        $activeOnTheFlyUpdate = false;
+    }
+}
+
+    if ($entity_model::isColumnNullable($field['name'])) {
+        $allows_null = isset($field['allows_null']) ? $field['allows_null'] : true;
+    }else {
+        $allows_null = isset($field['allows_null']) ? $field['allows_null'] : false;
     }
+
 @endphp
 
 <div @include('crud::inc.field_wrapper_attributes') >
 
     <label>{!! $field['label'] !!}</label>
     @include('crud::inc.field_translatable_icon')
-
-    <select
+    @if($activeOnTheFlyCreate)
+        @include('crud::buttons.on_the_fly.create', ['name' => $field['name'], 'onTheFlyEntity' => $onTheFlyEntity])
+       @endif
+<select
         name="{{ $field['name'] }}"
+        data-original-name="{{ $field['name'] }}"
         style="width: 100%"
         data-init-function="bpFieldInitSelect2Element"
+        data-is-on-the-fly="{{ $onTheFly ?? 'false' }}"
+        data-field-related-name="{{$onTheFlyEntity}}"
+        data-on-the-fly-create-route="{{$createRoute ?? false}}"
+        data-on-the-fly-update-route="{{$updateRoute ?? false}}"
+        data-on-the-fly-refresh-route="{{$refreshRoute ?? false}}"
+        data-field-multiple="false"
+        data-on-the-fly-related-key="{{$entity_model_instance->getKeyName()}}"
+        data-on-the-fly-related-attribute="{{$field['attribute']}}"
+        data-options-for-select="{{json_encode($options)}}"
+        data-on-the-fly-create-button="{{ $onTheFlyEntity }}-on-the-fly-create-{{$field['name']}}"
+        data-on-the-fly-allow-create="{{var_export($activeOnTheFlyCreate)}}"
+        data-on-the-fly-allow-update="{{var_export($activeOnTheFlyUpdate)}}"
+        data-allows-null="{{var_export($allows_null)}}"
+        data-current-value="{{$current_value}}"
         @include('crud::inc.field_attributes', ['default_class' =>  'form-control select2_field'])
         >
-
-        @if ($entity_model::isColumnNullable($field['name']))
-            <option value="">-</option>
-        @endif
-
-        @if (count($options))
-            @foreach ($options as $option)
-                @if($current_value == $option->getKey())
-                    <option value="{{ $option->getKey() }}" selected>{{ $option->{$field['attribute']} }}</option>
-                @else
-                    <option value="{{ $option->getKey() }}">{{ $option->{$field['attribute']} }}</option>
-                @endif
-            @endforeach
-        @endif
     </select>
 
     {{-- HINT --}}
     @if (isset($field['hint']))
         <p class="help-block">{!! $field['hint'] !!}</p>
     @endif
+
 </div>
 
 {{-- ########################################## --}}
 {{-- Extra CSS and JS for this particular field --}}
 {{-- If a field type is shown multiple times on a form, the CSS and JS will only be loaded once --}}
+
+@if (!$crud->isAnyTypeLoaded([
```

Comment:
I think we should use this approach in conjunction with fieldTypeLoaded. Some things we just want to incluce ONCE for multiple field types.